src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,811 @@
+        return _maxMagnitudeHistory[operator][strategy].latest() - curEncumberedMagnitude;
+    }
+
+    /// @inheritdoc IAllocationManager
+    function getMaxMagnitude(address operator, IStrategy strategy) public view returns (uint64) {
+        return _maxMagnitudeHistory[operator][strategy].latest();
+    }
+
+    /// @inheritdoc IAllocationManager
+    function getMaxMagnitudes(
+        address operator,
+        IStrategy[] memory strategies
+    ) external view returns (uint64[] memory) {
+        uint64[] memory maxMagnitudes = new uint64[](strategies.length);
+
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            maxMagnitudes[i] = getMaxMagnitude(operator, strategies[i]);
+        }
+
+        return maxMagnitudes;
+    }
+
+    /// @inheritdoc IAllocationManager
+    function getMaxMagnitudes(address[] memory operators, IStrategy strategy) external view returns (uint64[] memory) {
+        uint64[] memory maxMagnitudes = new uint64[](operators.length);
+
+        for (uint256 i = 0; i < operators.length; ++i) {
+            maxMagnitudes[i] = getMaxMagnitude(operators[i], strategy);
+        }
+
+        return maxMagnitudes;
+    }
+
+    /// @inheritdoc IAllocationManager
+    function getMaxMagnitudesAtBlock(
+        address operator,
+        IStrategy[] memory strategies,
+        uint32 blockNumber
+    ) external view returns (uint64[] memory) {
+        uint64[] memory maxMagnitudes = new uint64[](strategies.length);
+
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            maxMagnitudes[i] = _maxMagnitudeHistory[operator][strategies[i]].upperLookup({key: blockNumber});
+        }
+
+        return maxMagnitudes;
+    }
+
+    /// @inheritdoc IAllocationManager
+    function getAllocationDelay(
+        address operator
+    ) public view returns (bool, uint32) {
+        AllocationDelayInfo memory info = _allocationDelayInfo[operator];
+
+        uint32 delay = info.delay;
+        bool isSet = info.isSet;
+
+        // If there is a pending delay that can be applied, apply it
+        if (info.effectBlock != 0 && block.number >= info.effectBlock) {
+            delay = info.pendingDelay;
+            isSet = true;
+        }
+
+        return (isSet, delay);
+    }
+
+    /// @inheritdoc IAllocationManager
+    function getRegisteredSets(
+        address operator
+    ) public view returns (OperatorSet[] memory) {
+        uint256 length = registeredSets[operator].length();
+        OperatorSet[] memory operatorSets = new OperatorSet[](length);
+
+        for (uint256 i = 0; i < length; ++i) {
+            operatorSets[i] = OperatorSetLib.decode(registeredSets[operator].at(i));
+        }
+
+        return operatorSets;
+    }
+
+    /// @inheritdoc IAllocationManager
+    function isOperatorSet(
+        OperatorSet memory operatorSet
+    ) external view returns (bool) {
+        return _operatorSets[operatorSet.avs].contains(operatorSet.id);
+    }
```

Comment:
Can we add 

isMemeberOfOperatorSet(address operator, OperatorSet memory operatorSet)